### PR TITLE
Convert GameList col enum to enum class

### PR DESCRIPTION
**Pull request by Dentomologist**
Also adds a couple lambdas and using directives to reduce visual noise.

### CODE REVIEW

File: Source/Core/DolphinQt/GameList/GameListModel.h [hunk 1]
```diff
@@ -44,23 +44,23 @@ class GameListModel final : public QAbstractTableModel
...
+    ID,
+    Country,
+    Size,
+    FileName,
+    FilePath,
+    FileFormat,
+    BlockSize,
+    Compression,
+    Tags,
+    Count
```
Comment (Rumi-Larry) [inline Source/Core/DolphinQt/GameList/GameListModel.h:63]: Extra comma for future diffs

File: Source/Core/DolphinQt/GameList/GameList.cpp [hunk 2]
```diff
@@ -145,34 +145,42 @@ void GameList::MakeListView()
...
-  hor_header->setSectionResizeMode(GameListModel::COL_COMPRESSION, QHeaderView::Fixed);
-  hor_header->setSectionResizeMode(GameListModel::COL_TAGS, QHeaderView::Interactive);
+    m_list->sortByColumn(static_cast<int>(GameListModel::Column::Title), Qt::AscendingOrder);
+
+  const auto SetResizeMode = [&hor_header](const GameListModel::Column column,
+                                           const QHeaderView::ResizeMode mode) {
+    hor_header->setSectionResizeMode(static_cast<int>(column), mode);
+  };
+  {
+    using column = GameListModel::Column;
```
Comment (lioncash) [inline Source/Core/DolphinQt/GameList/GameList.cpp:155]: Given this is a type alias, it should still be cased like `This`

File: Source/Core/DolphinQt/GameList/GameList.cpp [hunk 3]
```diff
@@ -192,26 +200,26 @@ GameList::~GameList()
...
-                          !SConfig::GetInstance().m_showBlockSizeColumn);
-  m_list->setColumnHidden(GameListModel::COL_COMPRESSION,
-                          !SConfig::GetInstance().m_showCompressionColumn);
-  m_list->setColumnHidden(GameListModel::COL_TAGS, !SConfig::GetInstance().m_showTagsColumn);
+  const auto& config = SConfig::GetInstance();
+  const auto SetVisiblity = [this](const GameListModel::Column column, const bool is_visible) {
+    m_list->setColumnHidden(static_cast<int>(column), !is_visible);
+  };
+
+  using column = GameListModel::Column;
```
Comment (lioncash) [inline Source/Core/DolphinQt/GameList/GameList.cpp:208]: Ditto

File: Source/Core/DolphinQt/GameList/GameList.cpp [hunk 4]
```diff
@@ -881,23 +889,23 @@ void GameList::keyPressEvent(QKeyEvent* event)
...
-      {tr("Game ID"), GameListModel::COL_ID},
-      {tr("Region"), GameListModel::COL_COUNTRY},
-      {tr("File Size"), GameListModel::COL_SIZE},
-      {tr("File Format"), GameListModel::COL_FILE_FORMAT},
-      {tr("Block Size"), GameListModel::COL_BLOCK_SIZE},
-      {tr("Compression"), GameListModel::COL_COMPRESSION},
-      {tr("Tags"), GameListModel::COL_TAGS}};
-
-  m_list->setColumnHidden(rowname_to_col_index[row], !visible);
+  using column = GameListModel::Column;
```
Comment (lioncash) [inline Source/Core/DolphinQt/GameList/GameList.cpp:892]: Ditto

File: Source/Core/DolphinQt/GameList/GameList.cpp [hunk 5]
```diff
@@ -881,23 +889,23 @@ void GameList::keyPressEvent(QKeyEvent* event)
...
+                                                          {tr("Maker"), column::Maker},
+                                                          {tr("File Name"), column::FileName},
+                                                          {tr("File Path"), column::FilePath},
+                                                          {tr("Game ID"), column::ID},
+                                                          {tr("Region"), column::Country},
+                                                          {tr("File Size"), column::Size},
+                                                          {tr("File Format"), column::FileFormat},
+                                                          {tr("Block Size"), column::BlockSize},
+                                                          {tr("Compression"), column::Compression},
+                                                          {tr("Tags"), column::Tags}};
```
Comment (lioncash) [inline Source/Core/DolphinQt/GameList/GameList.cpp:906]: Adding a comma after the last entry in the map should make this format a little better automatically.

File: Source/Core/DolphinQt/GameList/GameList.cpp (hunk 2 above)
Comment (Pokechu22) [reply Source/Core/DolphinQt/GameList/GameList.cpp]: Pretty sure it could be just `using GameListModel::Column` in that case.

Comment (Dentomologist) [reply Source/Core/DolphinQt/GameList/GameList.cpp]: [Unfortunately not](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-2/compiler-error-c2885?view=msvc-160).

File: Source/Core/DolphinQt/GameList/GameList.cpp [hunk 6]
```diff
@@ -145,34 +145,42 @@ void GameList::MakeListView()
...
-  hor_header->setSectionResizeMode(GameListModel::COL_TAGS, QHeaderView::Interactive);
+    m_list->sortByColumn(static_cast<int>(GameListModel::Column::Title), Qt::AscendingOrder);
+
+  const auto SetResizeMode = [&hor_header](const GameListModel::Column column,
+                                           const QHeaderView::ResizeMode mode) {
+    hor_header->setSectionResizeMode(static_cast<int>(column), mode);
+  };
+  {
+    using Column = GameListModel::Column;
+    using mode = QHeaderView::ResizeMode;
```
Comment (BhaaLseN) [inline Source/Core/DolphinQt/GameList/GameList.cpp:156]: Did you leave this lower-cased on purpose while changing `Column` to sentence case?

Comment (Dentomologist) [reply Source/Core/DolphinQt/GameList/GameList.cpp:156]: Nope, I just missed it.

**Review by leoetlino — APPROVED** [review]

**Pull request merged**